sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/controller/UserController.java

```diff
@@ -632,4 +636,79 @@
           String id) {
     return getUserByUsername(realm, null, id);
   }
+
+  @GetMapping(
+      path = {"/realms/{realm}/storages/{storage}/users/mail/{mail}"},
+      produces = {MediaType.APPLICATION_JSON_VALUE})
+  @Operation(summary = "Get user by mail")
+  @ApiResponses(
+      value = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "User found according to parameters",
+            content = {
+              @Content(
+                  mediaType = "application/json",
+                  schema = @Schema(implementation = User.class))
+            })
+      })
+  @PreAuthorize("@NewAuthorizeMethodDecider.isReader(#realm,#storage)")
+  public ResponseEntity<User> getUserByMail(
+      @Parameter(
+              description = "Name of the realm where the operation will be made",
+              required = true)
+          @PathVariable("realm")
+          String realm,
+      @Parameter(
+              description = "Name of the userStorage where the operation will be made",
+              required = false)
+          @PathVariable(name = "storage", required = false)
+          String storage,
+      @Parameter(description = "User's mail to search", required = true) @PathVariable("mail")
+          String mail) {
+    if (Boolean.parseBoolean(
+        realmService
+            .load(realm)
+            .orElseThrow(() -> new RealmNotFoundException("Realm " + realm + " doesn't exist"))
+            .getProperties()
+            .get(GlobalKeysConfig.VERIFY_MAIL_UNICITY))) {
+
+      User user =
+          userService
+              .findByMail(realm, storage, mail)
+              .orElseThrow(
+                  () ->
+                      new UserNotFoundException(
+                          "Cannot find user with mail " + mail + " in realm " + realm));
+      return ResponseEntity.status(HttpStatus.OK).body(user);
+    }
+    return ResponseEntity.status(HttpStatus.BAD_GATEWAY).build();
```

Comment:
```suggestion
    return ResponseEntity.status(HttpStatus.NOT_IMPLEMENTED).build();
```

This seems better and is already used elsewhere